src/validation.cpp

```diff
@@ -1148,14 +1148,18 @@ bool MemPoolAccept::SubmitPackage(const ATMPArgs& args, std::vector<Workspace>&
     // make sure we haven't exceeded max mempool size.
     LimitMempoolSize(m_pool, m_active_chainstate.CoinsTip());
 
+    std::vector<uint256> all_package_wtxids;
+    std::transform(workspaces.cbegin(), workspaces.cend(), std::back_inserter(all_package_wtxids),
+                   [](const auto& ws) { return ws.m_ptx->GetWitnessHash(); });
     // Find the wtxids of the transactions that made it into the mempool. Allow partial submission,
     // but don't report success unless they all made it into the mempool.
     for (Workspace& ws : workspaces) {
         const auto effective_feerate = args.m_package_feerates ? ws.m_package_feerate :
             CFeeRate{ws.m_modified_fees, static_cast<uint32_t>(ws.m_vsize)};
         if (m_pool.exists(GenTxid::Wtxid(ws.m_ptx->GetWitnessHash()))) {
             results.emplace(ws.m_ptx->GetWitnessHash(),
-                MempoolAcceptResult::Success(std::move(ws.m_replaced_transactions), ws.m_vsize, ws.m_base_fees, effective_feerate));
+                MempoolAcceptResult::Success(std::move(ws.m_replaced_transactions), ws.m_vsize,
+                                             ws.m_base_fees, effective_feerate, all_package_wtxids));
```

Comment:
In 0d17b04977d6882e8dba6cf6c192a825a446f4ae "[validation] return wtxids of other transactions whose fees were used"

My understanding is that `effective_feerate` is computed using all of the package's txs when `args.m_package_feerates == true`, but only the workspace's single tx when that parameter is `false`. So I think it is incorrect to give `all_package_wtxids` in the latter case?

Reply:
We include `MempoolAcceptResult::m_effective_feerate` and `m_wtxids_fee_calculations` whenever the transaction is successful i.e. `ResultType::VALID`. I think it's still meaningful for single transactions, as it can include modified fees from prioritisetransaction. Alternatively we could return a separate result for modified fees.

> My understanding is that effective_feerate is computed using all of the package's txs when args.m_package_feerates == true, but only the workspace's single tx when that parameter is false. 

This is not 100% accurate - it's all of the transactions within `AcceptMultipleTransactions`, but that's not equivalent to the entire package. During a package validation (see `AcceptPackage`) we may split up a package into multiple groups for fee calculations (i.e. right now transactions can be submitted individually and with #26711 various subgroups). The hope here is for the caller to know exactly which transactions were included in the "package feerate" for a transaction.

Reply:
Right, what I mean is that there is a mismatch between `m_effective_feerate` and `m_wtxid_fee_calculations` depending on `args.m_package_feerates`. When that parameter is true, we would correctly report `m_effective_feerate` as the package feerate and `m_wtxids_fee_calculations` as all of the wtxids in the workspaces. But when it is false, `m_effective_feerate` is the modified feerate of just the single transaction in the workspace, whereas `m_wtxid_fee_calculations` is still all of the wtxids in all of the workspaces. That is what I think is incorrect. It should be the workspace's single wtxid, rather than all of them.

Reply:
Ohh I see, yeah that's a bit weird. Changed now.